src/controller.ts

```diff
@@ -252,19 +262,43 @@ export default function (
       },
     );
 
-    fastify.post<{ 
+    fastify.get<{ Querystring: { code: string; provider: string } }>(
+      '/is-unique-code-available',
+      {
+        schema: {
+          querystring: {
+            type: 'object',
+            properties: { code: { type: 'string' }, provider: { type: 'string' } },
+          },
+        },
+        config: {
+          rateLimit: {
+            max: 5,
+            timeWindow: '1 minute',
+          },
+        },
+      },
+      async (req, res) => {
+        const { code, provider } = req.query;
+        await licenseCodesService.isLicenseCodeAvailable(code, provider);
```

Comment:
This endpoint has a failure that could be annoying for the server. This use case throws if the code is redeemed or not found. I suggest catching here those errors and returning the correspondent 404 code. If not, a 500 is going to be thrown, which is not semantic and false, as there is no internal server error happening rather than a 404 that we are not handling properly.

Reply:
Done